engine/classes/ElggMenu.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * Elgg Menu
+ *
+ * @package    Elgg.Core
+ * @subpackage Navigation
+ * @since      2.2.0
+ */
+class ElggMenu {
```

Comment:
I wonder how this functionality could be built into ElggMenuBuilder in a BC way. Like if we added $name and $vars to the constructor, it would handle calling the hooks...